dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java

```diff
@@ -425,23 +452,44 @@ public void updateIndex(Context context) {
      */
     @Override
     public void updateIndex(Context context, boolean force) {
-        try {
-            Iterator<Item> items = null;
-            for (items = itemService.findAllUnfiltered(context); items.hasNext(); ) {
-                Item item = items.next();
-                indexContent(context, item, force);
-                //To prevent memory issues, discard an object from the cache after processing
-                context.uncacheEntity(item);
-            }
+        updateIndex(context, force, Constants.ITEM);
+        updateIndex(context, force, Constants.COLLECTION);
+        updateIndex(context, force, Constants.COMMUNITY);
+    }
 
-            List<Collection> collections = collectionService.findAll(context);
-            for (Collection collection : collections) {
-                indexContent(context, collection, force);
-            }
 
-            List<Community> communities = communityService.findAll(context);
-            for (Community community : communities) {
-                indexContent(context, community, force);
+    public void updateIndex(Context context, List<UUID> ids, boolean force, int type) {
+        if (type != Constants.ITEM) {
+            throw new RuntimeException("Only ITEM is supported in this mode - type founded: " + type);
```

Comment:
Would an IllegalArgumentException not make more sense here instead of a generic "RuntimeException" as this is what it really is. An illegal Argument was passed along.

Reply:
the code doesn't exist anymore